muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/base/input-contacts-autofill.tsx

```diff
@@ -62,19 +64,37 @@ export default class InputContactsAutofill extends React.Component<InputContacts
     this.onInputBlur = this.onInputBlur.bind(this);
     this.onInputFocus = this.onInputFocus.bind(this);
     this.setHeight = this.setHeight.bind(this);
+    this.setBodyMargin = this.setBodyMargin.bind(this);
     this.onDelete = this.onDelete.bind(this);
   }
   componentWillReceiveProps(nextProps: InputContactsAutofillProps){
     if (nextProps.selectedItems !== this.props.selectedItems){
       this.setState({selectedItems: nextProps.selectedItems})
     }
   }
+  
+  
+// Deprecated? 
+  
   setHeight(){
-    let fieldHeight = (this.refs["taginput"] as TagInput).getHeight();
+    let fieldHeight = (this.refs["taginput"] as TagInput).getInputHeight();
     if (fieldHeight !== this.state.fieldHeight){
       this.setState({fieldHeight});
     }
   }
+  
+  setBodyMargin() {
+    let selectedHeight = (this.refs["taginput"] as TagInput).getSelectedHeight();
+    let prevSelectedHeight= this.state.selectedHeight;
+    let currentBodyMargin = parseFloat(document.body.style.marginBottom);       
+    let defaultBodyMargin = currentBodyMargin - prevSelectedHeight;
+    
+    if (selectedHeight !== this.state.selectedHeight){
+      let bodyMargin = defaultBodyMargin + selectedHeight + "px";        
+        document.body.style.marginBottom = bodyMargin;
+        this.setState({selectedHeight: selectedHeight});
```

Comment:
Using a setState call causes the component to recheck whether it needs to rerender or recalculate its properties, this can be expensive; you can declare a private number instead the same way there's a NodeJS.Timer private variable in the function, you are not using the selectedHeight to modify the component but rather to modify the body which means the state doesn't really belong to the component; while it works it is cheaper to store this as an attribute of the class rather than an attribute of the state.

So as:

`if (newSelectedHeigth !== this.selectedHeight){`

and

`this.selectedHeight = newSelectedHeight`

If you see the fieldHeight because it was part of the state that's because fieldHeight was used for pixels offset, it affected the component visuals (BTW you need to remove that too since we are dropping setHeight)